pytest/tests/mocknet/outage.py

```diff
@@ -0,0 +1,55 @@
+# Stop fewer than one third of mocknet nodes for two full epochs.
+# During this time the network should continue to function.
+# The stopped nodes are then restarted, should come back up,
+# sync successfully and become operational again.
+
+import sys, time
+from rc import pmap
+
+sys.path.append('lib')
+import mocknet
+
+nodes = mocknet.get_nodes()
+epoch_length = mocknet.get_epoch_length_in_seconds(nodes[0])
+
+# Nodes 0, 1, 2, and 3 are validators and all have equal stake.
+# We stop node 0, thus stopping 25% of the stake.
+mocknet.stop_node(nodes[0])
+
+# Function to do a transfer, but not using the first node
+check_transfer = lambda: mocknet.transfer_between_nodes(nodes[1:])
+
+# Wait a moment
+time.sleep(1)
+
+# Check the network still functions
+check_transfer()
+
+# Wait an epoch
+time.sleep(epoch_length)
```

Comment:
if we want to accurately do this it is better to not rely on time. You can use the validator rpc https://docs.near.org/docs/interaction/rpc#validators to get the epoch start height and see whether that has changed.

Reply:
Thanks for pointing out this rpc to me!

I've improved the outage test by looking at block height for epoch changes, as you suggested. I also noticed that the offline node gets kicked out of the validator set, so I added some assertions around that as well as made sure the validator rejoins at the end of the test so that the network is left in the same state as when the test began.